server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadReportAnalysisMetadataHolderStep.java

```diff
@@ -39,12 +44,31 @@ public LoadReportAnalysisMetadataHolderStep(BatchReportReader reportReader, Muta
   @Override
   public void execute() {
     BatchReport.Metadata reportMetadata = reportReader.readMetadata();
+
+    checkProjectKeyConsistency(reportMetadata);
+
     mutableAnalysisMetadataHolder.setRootComponentRef(reportMetadata.getRootComponentRef());
     mutableAnalysisMetadataHolder.setBranch(reportMetadata.hasBranch() ? reportMetadata.getBranch() : null);
     mutableAnalysisMetadataHolder.setAnalysisDate(reportMetadata.getAnalysisDate());
     mutableAnalysisMetadataHolder.setCrossProjectDuplicationEnabled(reportMetadata.hasCrossProjectDuplicationActivated() && reportMetadata.getCrossProjectDuplicationActivated());
   }
 
+  private void checkProjectKeyConsistency(BatchReport.Metadata reportMetadata) {
+    String reportProjectKey = projectKeyFromReport(reportMetadata);
+    checkState(
+      ceTask.getComponentKey().equals(reportProjectKey),
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'ceTask' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)